### PR TITLE
Refactor from tag to role and fix many issues 

### LINKED ISSUE #87: Unable to change CCA label using edit command
If I want to change the cca labeling using the edit command, it won't work. But others work well.
image

### LINKED ISSUE #89: Both edit t/tag and assign command could change the role
Screenshot 2024-03-28 at 16 11 10
Screenshot 2024-03-28 at 16 12 50

So, I'm not sure whether to keep this feature for simplicity or fix it.

### LINKED ISSUE #90: Internally change Tags to Roles
This is for consistency purposes and its driving me insane because I keep trying to recall if Tags refer to Roles or CCA

### PR DESCRIPTION
**Pull request by JuliaPoo**
Mainly fixes #90, coincidentally fixes #89 as it removes the `/t` argument entirely.

1. Fixed `edit` command not being able to edit CCAs only. Fixes #87 
2. Fixed `Role`s not being able to have spaces (contradicts examples in the UG)
2. Fixed `edit` command example is malformed
3. Wrong `assign` documentation
4. A bunch of string builders were missing CCA field
    - Also fixed the serialising tests to include the CCA field
5. A bunch of code that is `Cca` object adjacent has documentation that mentions `tag` instead of `cca`.

For those testing this out, remember to remove your `addressbook.json` file and let it be regenerated as this pull changes the json format.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 1]
```diff
@@ -122,11 +122,11 @@ public boolean isAnyFieldNotEdited() {
...
 
-        public void setRole(Set<Tag> role) {
+        public void setRole(Set<Role> role) {
             this.role = role;
         }
 
-        public Optional<Set<Tag>> getRole() {
+        public Optional<Set<Role>> getRole() {
             return (role != null) ? Optional.of(Collections.unmodifiableSet(role)) : Optional.empty();
         }
```
Comment (LifHoshi) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java:132]: OK for this

File: src/main/java/seedu/address/model/roles/Role.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    /**
+     * Format state as text for viewing.
+     */
+    public String toString() {
+        return '[' + roleName + ']';
+    }
+
+}
```
Comment (LifHoshi) [inline src/main/java/seedu/address/model/roles/Role.java:62]: Compared with the original tag file, no issue here. But better for someone else look at it again

File: docs/UserGuide.md [hunk 3]
```diff
@@ -415,12 +415,12 @@ _Details coming soon ..._
...
+**Add**    | `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [r/ROLE] [c/CCA]…​` <br> e.g., `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 r/friend r/colleague c/CCA Cycling`
 **Clear**  | `clear`
 **Delete** | `delete INDEX`<br> e.g., `delete 3`
-**Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
+**Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [r/ROLE] [c/CCA]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
 **Find**   | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`
 **List**   | `list`
 **Filter** | `filter CCA1, CCA2, ...` e.g. `filter NUS Cycling`
-**Assign** | `Assign INDEX /r ROLE` e.g. `Assign 2 /r Member`
+**Assign** | `assign INDEX r/ROLE` e.g. `assign 2 r/Member`
```
[the remark this answers is not in the document]
Comment (JuliaPoo) [reply docs/UserGuide.md]: Your fix doesn't really fix it as `Assign` should be `assign`. So I'm fixing it tgt in this branch.

File: docs/diagrams/StorageClassDiagram.puml [hunk 4]
```diff
@@ -39,7 +39,7 @@ JsonUserPrefsStorage .up.|> UserPrefsStorage
 JsonAddressBookStorage .up.|> AddressBookStorage
 JsonAddressBookStorage ..> JsonSerializableAddressBook
 JsonSerializableAddressBook --> "*" JsonAdaptedPerson
-JsonAdaptedPerson --> "*" JsonAdaptedTag
+JsonAdaptedPerson --> "*" JsonAdaptedRole
```
[the remark this answers is not in the document]
Comment (JuliaPoo) [reply docs/diagrams/StorageClassDiagram.puml:42]: Can you elaborate on this?

File: src/main/java/seedu/address/model/roles/Role.java (hunk 2 above)
Comment (JuliaPoo) [reply src/main/java/seedu/address/model/roles/Role.java:62]: Okay, do you suggest we wait for somebody else to look through these changes, or do I merge it now?

File: docs/diagrams/StorageClassDiagram.puml (hunk 4 above)
[the remark this answers is not in the document]
Comment (LifHoshi) [reply docs/diagrams/StorageClassDiagram.puml:42]: This is the last file for documentation, so I wrote this comment. After this file is the code part. If no conflict above then can ignore

File: src/main/java/seedu/address/model/roles/Role.java (hunk 2 above)
Comment (LifHoshi) [reply src/main/java/seedu/address/model/roles/Role.java:62]: Better for another guy look through it again

**Review by Fidget-Spinner — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 5]
```diff
@@ -141,29 +141,29 @@ public static class EditPersonDescriptor {
...
+            setRoles(toCopy.roles);
             setCcas(toCopy.ccas);
         }
 
         /**
          * Returns true if at least one field is edited.
          */
         public boolean isAnyFieldEdited() {
-            return CollectionUtil.isAnyNonNull(name, phone, email, address, tags);
+            return CollectionUtil.isAnyNonNull(name, phone, email, address, ccas, roles);
```
Comment (JuliaPoo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:166]: Here's where I fixed "Fixed edit command not being able to edit CCAs only."

File: src/main/java/seedu/address/model/roles/Role.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Role in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidRoleName(String)}
+ */
+public class Role {
+
+    public static final String MESSAGE_CONSTRAINTS = "Role names should be alphanumeric with whitespace";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}\\s]+";
```
Comment (JuliaPoo) [inline src/main/java/seedu/address/model/roles/Role.java:13]: Here's where I fixed "Fixed Roles not being able to have spaces (contradicts examples in the UG)"

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 7]
```diff
@@ -45,7 +45,7 @@ public class EditCommand extends Command {
             + "[" + PREFIX_PHONE + "PHONE] "
             + "[" + PREFIX_EMAIL + "EMAIL] "
             + "[" + PREFIX_ADDRESS + "ADDRESS] "
-            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_ROLE + "ROLE] "
```
Comment (JuliaPoo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:48]: Here's where I fixed "Fixed edit command example is malformed"

File: docs/UserGuide.md [hunk 8]
```diff
@@ -415,12 +415,12 @@ _Details coming soon ..._
...
+**Add**    | `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [r/ROLE] [c/CCA]…​` <br> e.g., `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 r/friend r/colleague c/CCA Cycling`
 **Clear**  | `clear`
 **Delete** | `delete INDEX`<br> e.g., `delete 3`
-**Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
+**Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [r/ROLE] [c/CCA]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
 **Find**   | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`
 **List**   | `list`
 **Filter** | `filter CCA1, CCA2, ...` e.g. `filter NUS Cycling`
-**Assign** | `Assign INDEX r/ ROLE` e.g. `Assign 2 r/ Member` or `Assign 2 r/Member`
+**Assign** | `assign INDEX r/ROLE` e.g. `assign 2 r/ Member` or `assign 2 r/Member`
```
Comment (JuliaPoo) [inline docs/UserGuide.md:425]: Here's where I fixed "Wrong assign documentation"

**Pull request merged**